### PR TITLE
Add ability to Test feeds and AJAX load the log when doing manual pull

### PR DESCRIPTION
**Pull request by sc0ttkclark**
screen shot 2015-01-30 at 10 25 10 pm

And on clicking of Manual pull, it now AJAX refreshes the log meta box with the latest from the manual pull.

I also fixed an issue with the meta box #fp_manual_pull that the JS was targeting unintentionally for the button with the same ID. Changed button ID to #fp_manual_pull_run

### CODE REVIEW

**Comment by sc0ttkclark** [comment]
I should also note that to be able to do all this, I had to abstract the feed fetching functionality into it's own `get_feed_posts` method and give it public access. `do_pull` still has private access and utilizes this new abstracted method just fine.

The constructor now accepts a false, and if false it will not run `do_pull` (null still runs all feeds).

**Comment by sc0ttkclark** [comment]
This does not send the current feed settings from the form, it still requires saving, just like Do Feed Pull. I chose not to change that functionality there.

File: includes/class-fp-ajax.php [hunk 1]
```diff
@@ -52,15 +53,28 @@ public function action_get_namespaces() {
 	 */
 	public function action_pull() {
 		$output = array();
+		$output['message'] = __( 'Invalid AJAX Request, try again.', '' );
```
Comment (tlovett1) [inline includes/class-fp-ajax.php]: Can you add feed-pull as the domain?

File: includes/class-fp-ajax.php [hunk 2]
```diff
@@ -68,6 +82,56 @@ public function action_pull() {
 	}
 
 	/**
+	 * Do a feed pull test
+	 *
+	 * @since 0.1.0
```
Comment (tlovett1) [inline includes/class-fp-ajax.php]: We are going to call the next version 1.0

File: includes/class-fp-pull.php [hunk 3]
```diff
@@ -134,15 +138,101 @@ private function handle_feed_log( $source_feed_id ) {
 	}
 
 	/**
-	 * Pull from all our source feeds
+	 * Get source feed posts
 	 *
 	 * @param int $source_feed_id
+	 * @since 0.2.4
```
Comment (tlovett1) [inline includes/class-fp-pull.php]: 1.0

**Comment by sc0ttkclark** [comment]
Going to close and rebased (w/ changes as requested) on develop

**Pull request closed without merging**